File: backend/app/schemas/institutional.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import date

# Holding Schema
class FundHoldingBase(BaseModel):
    ticker: str
    shares: float
    value: float
    percent_portfolio: Optional[float] = None
    date_reported: Optional[date] = None

class FundHolding(FundHoldingBase):
    id: int
    fund_id: int

    class Config:
        from_attributes = True

# Fund Schema
class InstitutionalFundBase(BaseModel):
    name: str
    manager: Optional[str] = None
    cik: str
    total_assets: Optional[float] = None
    filing_date: Optional[date] = None
    image_url: Optional[str] = None

class InstitutionalFund(InstitutionalFundBase):
    id: int
    holdings: List[FundHolding] = []

    class Config:
        from_attributes = True

# Analytics Schemas
class TopMover(BaseModel):
    ticker: str
    total_value: float
    fund_count: int

class PortfolioStats(BaseModel):
    total_holdings_count: int
    top_holdings: List[dict] # simplified dict for now

File: frontend/src/types/institutional.ts
export interface FundHolding {
    id: number;
    fund_id: number;
    ticker: string;
    shares: number;
    value: number;
    percent_portfolio?: number;
    date_reported?: string; // API returns ISO string for date
}

export interface InstitutionalFund {
    id: number;
    name: string;
    manager?: string;
    cik: string;
    total_assets?: number;
    filing_date?: string;
    image_url?: string;
    holdings: FundHolding[];
}

export interface TopMover {
    ticker: string;
    total_value: number;
    fund_count: number;
}

export interface PortfolioStats {
    total_holdings_count: number;
    top_holdings: {
        ticker: string;
        percent: number;
        value: number;
    }[];
}
